libsrc/lynx/ser/lynx-comlynx.s

```diff
@@ -308,7 +288,7 @@ PutByte:
 ; Must return an SER_ERR_xx code in a/x.
 
 SER_STATUS:
-        ldy     SerialStat
+        lda     SerialStat
         ldx     #$00
         sta     (ptr1,x)
```

Comment:
This opcode seems to be weird, it works, but rather use sta (ptr1)

Reply:
I assume that this is done to express that we are setting member zero of structure ptr (which is an address). It seems to be for clarity rather than brevity.
Agree to leave?

Reply:
sta (nn,x) != sta (nn),y !!!

First: Use address at nn+x.
So if nn == 10 and x and y are 2, first will use the address at 12 to store to, and second will store to address in 12.

So, no, better remove.

Reply:
sta (ptr1) also saves two bytes, no? :)

Reply:
Here is the prototype for ser_status:
```
unsigned char __fastcall__ ser_status (unsigned char* status);
```
As far as I knew the calling convention assumes that the return value is in A and X should be 0. 
ptr1 that corresponds to the argument list. The first argument in the list is `status` (of type `unsigned char*`). By using Absolute Indexed Indirect addressing `sta (ptr1,x)` we essentialy save to the same location as sta (ptr1) does. 
No problem for me to change it into sta(ptr1). I left the original code there as all implementations of ser_status use it like this. If it saves two bytes, why not? ;)

Reply:
All three version use 2 bytes only. But "sta (ptr1,x)" is simply wrong here.
And if you gonna change code, you should remove such.

Reply:
6502 has no sta(zp) - so i guess that is why other drivers use it. its a common pattern when X is 0 and Y can't be used for whatever reason.

Reply:
:-)
I see.

Luckily the Lynx is an 65C02.

Reply:
Made the change to sta (ptr1) in two places. The routine set_get also used the same construct.